File: TP1/ClienteBackground/Mensagem.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Mensagem.cpp
 * Author: gustavo
 * 
 * Created on September 16, 2017, 1:28 PM
 */

#include "Mensagem.h"

Mensagem::Mensagem() {
}

Mensagem::Mensagem(int codigo, string texto=""){
    this->codigo = codigo;
    this->texto = texto;
}

Mensagem::Mensagem(char* mensagem){
    char txt[255];
    sscanf(mensagem, "%d|%s", &this->codigo, txt);
    this->texto = *(new string(txt));
}

Mensagem::Mensagem(const Mensagem& orig) {
}

Mensagem::~Mensagem() {
}

void Mensagem::toChar(char* resultado){
    sprintf(resultado, "%d|%s", codigo, texto.data()); 
}

int Mensagem::getCodigo(){
    return this->codigo;
}
    
void Mensagem::setCodigo(int codigo){
    this->codigo = codigo;
}

string Mensagem::getTexto(){
    return this->texto;
}

void Mensagem::setTexto(string texto){
    this->texto = texto;
}

File: TP2-POC/src/tp2/poc/Mensagem.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tp2.poc;

import java.io.Serializable;

/**
 *
 * @author 123437
 */
public class Mensagem implements Serializable{
    
    private int codigo;
    private String texto;
    
    public Mensagem(){
        this.codigo = 1;
        this.texto = "Mensagem Teste";
    }

    public Mensagem(String mensagem){
        String[] partes = mensagem.split("|");
        this.codigo = Integer.parseInt(partes[0]);
        
        for(int i=1; i<partes.length; i++){
            this.texto = "|" + partes[i];
        }
        
        this.texto = this.texto.substring(1);
    }
    
    public Mensagem(int codigo, String texto){
        this.codigo = codigo;
        this.texto = texto;
    }
    
    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getTexto() {
        return texto;
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }

    @Override
    public String toString() {
        return this.codigo + "|" + this.texto;
    }
    
    public static Mensagem mensagemSolicitacaoPorta(){
        return new Mensagem(2, "");
    }
    
    public static Mensagem mensagemRespostaSolicitacaoPorta(int porta){
        return new Mensagem(3, Integer.toString(porta));
    }
}
